### DISCUSSION TITLE
Cartographie - API routing et de cartographie

### DISCUSSION THREAD

**Discussion by AdrienDSlone** [AdrienDSlone/slowTripPlanner#21] [4. Front]
Voici une compilation des API et outils disponibles, personnellement, je recommande GraphHopper :

GraphHopper Directions API with Route Optimization
https://www.graphhopper.com/

Documentation - GraphHopper Directions API
https://docs.graphhopper.com/

Leaflet - a JavaScript library for interactive maps
https://leafletjs.com/

Leaflet Routing Machine
https://www.liedman.net/leaflet-routing-machine/

Geovelo
https://geovelo.fr/fr/rides?c=1.500550%2C47.008249&max-distance=100000&z=5.56

OpenCycleMap.org - the OpenStreetMap Cycle Map
https://www.opencyclemap.org/

**Comment by AdrienDSlone** [comment]
Je recommande GraphHopper car elle est utilisé dans le projet :
https://github.com/tboegi/GpsMaster
![image](https://user-images.githubusercontent.com/50139099/172052610-ff1b21a4-0523-459b-bebd-1e2bd622d6ba.png)

de plus elle est gratuite :
https://www.graphhopper.com/pricing/
![image](https://user-images.githubusercontent.com/50139099/172052653-8c2c0fad-0fbd-4024-9ab3-052f431b1750.png)

https://www.graphhopper.com/developers/

**Reply by Scabiei** [reply]
Tu sais ce qu'ils veulent dire par "max locations per request : 5 locations" ?

J'ai tenté de lancé le projet GpsMaster qui l'utilise mais ça ne marche pas. J'aurais voulu testé si on pouvait bien mettre plus de 5 étapes sans être bloqué avec la version gratuite.

**Reply by AdrienDSlone** [reply]
Non je l'ignore, j'imagine que ça signifie 5 adresses IP ?

Pour créer des étapes il faut cliquer sur :
![image](https://user-images.githubusercontent.com/50139099/172801827-f500b6e6-13f0-4fd0-888d-327d74d8b4c9.png)

et tu peux placer autant d'étapes que tu veux :
![image](https://user-images.githubusercontent.com/50139099/172801928-24ce79f9-2653-4fc9-a9db-cacaa3248a2a.png)

**Comment by AdrienDSlone** [comment]
Et de la documentation supplémentaire :
graphhopper/directions-api-js-client: JavaScript client for the GraphHopper Directions API
https://github.com/graphhopper/directions-api-js-client

directions-api-js-client/GraphHopperRouting.js at master · graphhopper/directions-api-js-client
https://github.com/graphhopper/directions-api-js-client/blob/master/src/GraphHopperRouting.js

directions-api-js-client/GraphHopperOptimization.js at master · graphhopper/directions-api-js-client
https://github.com/graphhopper/directions-api-js-client/blob/master/src/GraphHopperOptimization.js
